zcash_client_backend/proto/proposal.proto

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+syntax = "proto3";
+package cash.z.wallet.sdk.ffi;
+
+// A data structure that describes the inputs to be consumed and outputs to
+// be produced in a proposed transaction.
+message Proposal {
+    uint32 protoVersion = 1;
+    // ZIP 321 serialized transaction request
+    string transactionRequest = 2;
+    // The unique identifier and amount for each proposed input
+    repeated ProposedInput inputs = 3;
+    // The total value, fee amount, and change outputs of the proposed
+    // transaction
+    TransactionBalance balance = 4;
+    // The fee rule used in constructing this proposal
+    FeeRule feeRule = 6;
+    // The minimum number of confirmations required for transaction inputs to be selected.
+    // Each of the proposed inputs is know to have at least this many confirmations.
+    uint32 minConfirmations = 7;
+    // The target height for which the proposal was constructed
+    //
+    // The chain must contain at least this many blocks in order for the proposal to
+    // be executed.
+    uint32 minTargetHeight = 8;
+    // Returns the anchor height used in preparing the proposal.
+    //
+    // If, at the time that the proposal is executed, the anchor height
+    // required to satisfy / the minimum confirmation depth is less than this
+    // height, the proposal execution / API will return an error.
+    uint32 minAnchorHeight = 9;
```

Comment:
I considered saying that might be an overspecification to give both these heights (i.e. that it might be sufficient to specify the anchor depth at the time the proposal is executed, as long as the confirmation depth constraints are still met). However, on reflection I think that might leak information and also be more complicated.

Reply:
Yeah, I would like the proposals to be essentially stable.